File: src/game/entity/enemies/npcs/YellowBall.java
package game.entity.enemies.npcs;

import game.entity.models.Coordinates;
import game.entity.models.Direction;
import game.ui.panels.game.PitchPanel;
import game.utils.Paths;

import java.util.Arrays;
import java.util.List;

public class YellowBall extends IntelligentEnemy {
    public YellowBall() {
        super();
    }

    public YellowBall(Coordinates coordinates) {
        super(coordinates);
    }

    @Override
    protected String getBasePath() {
        return Paths.getEnemiesFolder() + "/yellow_ball/yellow_ball";
    }

    @Override
    protected List<Direction> getImageDirections() {
        return Arrays.asList(Direction.RIGHT, Direction.LEFT);
    }

    @Override
    public String[] getCharacterOrientedImages() {
        return new String[]{
                String.format("%s/yellow_ball/yellow_ball_%s_%d.png", Paths.getEnemiesFolder(), imageDirection.toString().toLowerCase(), 0),
                String.format("%s/yellow_ball/yellow_ball_%s_%d.png", Paths.getEnemiesFolder(), imageDirection.toString().toLowerCase(), 1),
                String.format("%s/yellow_ball/yellow_ball_%s_%d.png", Paths.getEnemiesFolder(), imageDirection.toString().toLowerCase(), 2),
                String.format("%s/yellow_ball/yellow_ball_%s_%d.png", Paths.getEnemiesFolder(), imageDirection.toString().toLowerCase(), 3)
        };
    }

    @Override
    public int getSize() {
        return PitchPanel.COMMON_DIVISOR * 2;
    }
}

File: src/game/entity/enemies/npcs/YellowBall.kt
package game.entity.enemies.npcs

import game.entity.EntityTypes
import game.entity.models.Coordinates
import game.entity.models.Direction
import game.ui.panels.game.PitchPanel
import game.utils.Paths.enemiesFolder

class YellowBall : IntelligentEnemy {
    constructor(id: Long) : this() {
        this.id = id
    }

    constructor() : super()
    constructor(coordinates: Coordinates?) : super(coordinates)

    override val entitiesAssetsPath: String get() ="$enemiesFolder/yellow_ball/yellow_ball"

    override fun getImageDirections(): List<Direction> = listOf(Direction.RIGHT, Direction.LEFT)

    override fun getCharacterOrientedImages(): Array<String> = Array(4) { index ->
        "$enemiesFolder/yellow_ball/yellow_ball_${imageDirection.toString().lowercase()}_$index.png"
    }

    override val size: Int
        get() = PitchPanel.COMMON_DIVISOR * 2

    override val type: EntityTypes
        get() = EntityTypes.YellowBall
}